### PR TITLE
Move SpanFilterBuilder and collaborators to otel package

### PR DESCRIPTION
**Pull request by breedx-splk**
No real changes, just a lift-and-shift.

### CODE REVIEW

File: splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/SpanFilterBuilder.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.splunk.rum;
+package io.opentelemetry.rum.internal;
```
Comment (mateuszrzeszutek) [inline splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/SpanFilterBuilder.java:17]: The `SpanFilterBuilder` is part of the public API; we can't just move it to another package without introducing a breaking change.
WDYT about copying the existing `SpanFilterBuilder` over to the otel package and leaving a shim in the splunk package, that'd simply call the otel filter builder methods?

Comment (breedx-splk) [reply splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/SpanFilterBuilder.java:17]: Yeah good catch. I missed the fact that it is "api surface".

Comment (breedx-splk) [reply splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/SpanFilterBuilder.java:17]: Ok, I copied it back and turned it into a wrapper. Added method `configureSpanFilter` on `SplunkRumBuilder` and deprecated the other method that takes the splunk class. LMK what you think.

File: sample-app/build.gradle.kts [hunk 2]
```diff
@@ -51,7 +51,7 @@ android {
     }
 }
 
-val otelVersion = "1.23.1"
+val otelVersion = "1.23.1e"
```
Comment (mateuszrzeszutek) [inline sample-app/build.gradle.kts:54]: e?
```suggestion
val otelVersion = "1.23.1"
```

File: splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java [hunk 3]
```diff
@@ -240,16 +240,30 @@ public SplunkRumBuilder setDeploymentEnvironment(String environment) {
     }
 
     /**
-     * Configures span data filtering.
+     * Configures span data filtering. This method is deprecated in favor of configureSpanFilter().
```
Comment (mateuszrzeszutek) [inline splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java:243]: Nit
```suggestion
     * Configures span data filtering. This method is deprecated in favor of {@link #configureSpanFilter(Consumer)}.
```

File: splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java [hunk 4]
```diff
@@ -240,16 +240,30 @@ public SplunkRumBuilder setDeploymentEnvironment(String environment) {
...
 
+    /**
+     * Configures span data filtering.
+     *
+     * @param configurer A function that will configure the passed {@link
+     *     io.opentelemetry.rum.internal.SpanFilterBuilder}.
+     * @return {@code this}
+     */
+    public SplunkRumBuilder configureSpanFilter(
+            Consumer<io.opentelemetry.rum.internal.SpanFilterBuilder> configurer) {
```
Comment (mateuszrzeszutek) [inline splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java:262]: In general I think deprecating the old method and adding a new is a great idea, but I think we should probably wait a bit before we actually do that - at least until we upstream the OTel parts. If we add this as it is right now, we'll have our public API depend on the `io.opentelemetry.rum.internal` package.

File: sample-app/build.gradle.kts (hunk 2 above)
Comment (breedx-splk) [reply sample-app/build.gradle.kts:54]: Sorry, clearly a screw up.

File: splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java (hunk 4 above)
Comment (breedx-splk) [reply splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java]: Yeah ok, fair enough...we don't want to have references to the internal package in our public API. I get it now. We're going to have to deal with the fact that EVERYTHING in otel is in internal at some point tho. 🤷🏻 

I removed the deprecation and removed the new configure method.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**